marklogic-data-hub/src/main/java/com/marklogic/hub/impl/FlowManagerImpl.java

```diff
@@ -344,21 +368,14 @@ protected void deleteDocumentsInDirectory(String directory) {
             logger.info(format("Deleting documents in directory '%s' in staging database", directory));
         }
         DatabaseClient stagingClient = hubConfig.newStagingClient();
-        try {
-            stagingClient.newServerEval().javascript(query).evalAs(String.class);
-        } finally {
-            stagingClient.release();
-        }
+        //Since client is getting reused in one UI, client.release() has to be removed
```

Comment:
@srinathgit  @ryanjdew @aebadirad  I think this is an argument here for a new "getStagingClient()" method instead of "newStagingClient()".  "new" implies that the caller should own the lifecycle of the DatabaseClient, and thus that the caller is free to call "release" on it - i.e. "Give me a new DatabaseClient, and don't worry, I'll release it when I decide I'm done with it". Whereas "get" implies - "Give me your staging DatabaseClient, instantiating it as needed. I recognize it as 'yours', and thus I won't call release on it."

Arguably, "newStagingClient" should have been called "getStagingClient" in a single-user context as well. But my main point is that "new" implies that a new object is being created and thus the caller owns it, whereas "get" implies that HubConfig owns the object being returned.

Reply:
I agree with you. "newStagingClient()" in HubConfigImpl returns a new client object every time. In HubConfigSession, it looks like the decision was made to reuse the client. I will leave this in place for now

Reply:
Yes, we may want to rename the function in the hubconfig interface (as part of another ticket) and leave it to the implementation to determine if it should create a new client every time. The reasoning behind not creating a new client every time in the session implementation is to avoid holding onto plain text credentials by build all the required clients at time of login.